src/config/mod.rs

```diff
@@ -313,10 +313,13 @@ impl Resource {
         // Port with unspecified address will bind to all network interfaces
         // so we have to check for all Port resources if they share the same
         // port.
-        for (key, port, protocol0) in unspecified {
+        for (key, address0, protocol0) in unspecified {
             for (resource, components) in resource_map.iter_mut() {
                 if let Resource::Port(address, protocol) = resource {
-                    if address.port() == port && &protocol0 == protocol {
+                    // IP addresses can either be v4 or v6.
+                    // Therefore we check if the ip version matches, the port matches and if the protocol (TCP/UDP) matches
+                    // when checking for equality.
+                    if &address0 == address && &protocol0 == protocol {
```

Comment:
Is this a change in semantics here? If I'm reading right, this goes from comparing just the port and protocol to comparing port, IP version, protocol, _and address_. We may want that, but the comment says differently.

Reply:
So, I'm wondering what the expected behavior is? 
If we have a resource with { "127.0.0.1", "8080", TCP} and another with {"0.0.0.0", "8080", TCP} - should we mark them as conflicting? Theres a test that verifies that we mark these as conflicting `vector config::resource_tests::unspecified_network_interface`.  If we should, then I ll revert my commit: https://github.com/vectordotdev/vector/pull/14187/commits/d35a094e7fce3e487158aed82e243057450002f1 and specifically check for just the ip version, port and protocol.

Reply:
I think you're right. Those aren't actually conflicting, as we can create both simultaneously. However, then the test that checks for that needs to be updated.